### PR TITLE
feat: split firmware and CLI file-picker folder memory

### PR DESCRIPTION
**Pull request by nerdCopter**
AI Generated pull-request

## Summary
Load Firmware (.hex/.uf2/.bin) and the CLI tab's save-to-file/load-from-file
pickers now remember separate last-used folders instead of sharing one.

Firmware `.hex`/`.uf2`/`.bin` files and CLI dumps live in different folders
for most users (a firmware-releases/downloads folder vs. a personal
backups/support folder). Sharing one remembered folder meant picking a
firmware file would silently redirect the next CLI save there too, and vice
versa — this splits them so each keeps its own last-used location.

`FileSystem.pickOpenFile`/`pickSaveFile` take an optional `pickerId`. On the
web it passes straight through as the File System Access API's `id` option,
which Chromium already remembers per id on its own. On the Tauri desktop
build, the native dialog has no equivalent, so the last directory per
`pickerId` is persisted via the existing `ConfigStorage` (localStorage) and
fed back as `defaultPath`. `dirName`/`joinPath` are cross-OS: POSIX (`/`),
Windows (`\`), and Windows drive roots (`C:\`) are all handled.

Only the Firmware Flasher's local-file load and the CLI tab's save/load
pickers are touched. Other pickers (Presets, Logging, VTX, OSD font,
AutoBackup, LogoManager, blackbox viewer) are unchanged.

## Test plan
- [x] `npm run lint` clean (includes `vue-tsc` typecheck)
- [x] `npx prettier --check` clean
- [x] `npx vitest run`: 1296 passed (108 files), including 9 new
      `FileSystem` picker-id tests covering the File System Access API `id`
      passthrough, Tauri per-`pickerId` folder memory, and POSIX/Windows
      filesystem-root edge cases
- [x] `npm run build` clean
- [x] Manual test — Linux: load a .hex from one folder, then use CLI
      save-to-file — each dialog opens in its own last-used folder
- [ ] Manual test — Windows: same check
- [ ] Manual test — macOS: same check

## Summary by CodeRabbit

* **New Features**
  * File picker dialogs now remember the last directory used for different file types, making repeated file selection and saving more convenient.
  * Firmware and CLI file operations use separate remembered locations.
  * File path handling now correctly preserves filesystem root directories.

* **Tests**
  * Added coverage for remembered picker directories, filesystem roots, browser file picker behavior, and file picker configuration validation.

### CODE REVIEW

File: src/js/FileSystem.js [hunk 1]
```diff
@@ -139,6 +140,48 @@ function baseName(path) {
     return parts[parts.length - 1] || String(path);
 }
 
+// Directory portion of a native path, in its original separator style.
+function dirName(path) {
+    const str = String(path);
+    const cut = Math.max(str.lastIndexOf("/"), str.lastIndexOf("\\"));
+    return cut >= 0 ? str.slice(0, cut) : "";
```
[the remark this answers is not in the document]
Comment (nerdCopter) [reply src/js/FileSystem.js]: @CodeRabbitAI — AI Generated comment

**Actionable**: Fixed in commit `920feb39`. `dirName` now special-cases a bare separator and a drive-letter root (`^[A-Za-z]:[/\\]`), so `/` and `C:\` keep their trailing separator instead of collapsing to `""` or `C:`. Added POSIX-root and Windows-drive-root tests in `test/js/fileSystem.test.js`.

**Comment by nerdCopter** [comment]
@CodeRabbitAI — AI Generated comment

**Actionable**: Fixed in commit `0fc5e9f6` (re: `src/js/FileSystem.js:322`). `pickOpenFile`/`pickSaveFile` now validate `pickerId` against the File System Access spec's rule (ASCII alphanumeric, `_`, `-`, ≤32 chars) before any platform dispatch, throwing consistently rather than letting the web and Tauri paths diverge. Added tests in `test/js/fileSystem.test.js` covering invalid characters, the 32-char boundary, allowed `-`/`_`, and an empty pickerId (treated as omitted, not invalid).

File: src/js/FileSystem.js [hunk 2]
```diff
@@ -139,6 +140,70 @@ function baseName(path) {
...
+}
+
+// The File System Access API's `id` option only accepts this shape and
+// throws `TypeError` otherwise; enforced here too so an invalid id fails the
+// same way on every platform instead of only in the browser.
+const PICKER_ID_PATTERN = /^[A-Za-z0-9_-]{1,32}$/;
+
+function assertValidPickerId(pickerId) {
+    if (pickerId && !PICKER_ID_PATTERN.test(pickerId)) {
+        throw new Error(`Invalid pickerId "${pickerId}": must be 1-32 ASCII letters, digits, "_" or "-".`);
```
[the remark this answers is not in the document]
Comment (nerdCopter) [reply src/js/FileSystem.js]: @coderabbitAI , resolved?

[the remark this answers is not in the document]
Comment (nerdCopter) [reply src/js/FileSystem.js]: @CodeRabbitAI — AI Generated comment

**Actionable**: Fixed in commit `be6913e8`. `assertValidPickerId` now throws `TypeError` instead of `Error`, matching the File System Access API's own contract for an invalid `id`. Updated the validation tests to assert `TypeError`.

**Comment by haslinghuis** [comment]
Nice can we cover all save / restore paths used in the app to replace #4001

**Comment by nerdCopter** [comment]
> Nice can we cover all save / restore paths used in the app to replace #4001

can it be a followup?  this solves my immediate annoyance for now.

**Comment by haslinghuis** [comment]
Please fix the sonar too

**Comment by nerdCopter** [comment]
AI Generated comment

Fixed in commit `1621c5da`: extracted `saveAs`/`openAs` helpers for the "pickerId remembers the last-used folder" tests and a shared `INVALID_PICKER_ID` matcher for the validation tests, removing the repeated mock/call/mock/call and TypeError/message assertion boilerplate SonarCloud flagged. Same assertions and coverage, no behavior change.

**Review by blckmn — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]
